kmip/tests/unit/services/server/test_engine.py

```diff
@@ -3461,6 +3462,91 @@ def test_request_omitting_secret(self):
             *args
         )
 
+    def test_locate(self):
+        """
+        Test that a Locate request can be processed correctly.
+        """
+        # TODO Need add more extensive tests after locate operaton is
+        # fully supported
+        e = engine.KmipEngine()
+        e._data_store = self.engine
+        e._data_store_session_factory = self.session_factory
+        e._data_session = e._data_store_session_factory()
+        e._logger = mock.MagicMock()
+
+        obj_a = pie_objects.OpaqueObject(b'', enums.OpaqueDataType.NONE)
+        obj_b = pie_objects.OpaqueObject(b'', enums.OpaqueDataType.NONE)
+
+        # locate should return nothing at beginning
+        payload = locate.LocateRequestPayload()
+        response_payload = e._process_locate(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Locate"
+        )
+        self.assertEqual(
+           len(response_payload.unique_identifiers),
+           0
+        )
+
+        # Add the first obj and test the locate
+        e._data_session.add(obj_a)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        id_a = str(obj_a.unique_identifier)
+
+        payload = locate.LocateRequestPayload()
+        e._logger.reset_mock()
+        response_payload = e._process_locate(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Locate"
+        )
+
+        self.assertEqual(
+           len(response_payload.unique_identifiers),
+           1
+        )
+        self.assertEqual(
+            id_a,
+            response_payload.unique_identifiers[0].value
+        )
+
+        # Add the second obj and test the locate
+        e._data_session.add(obj_b)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        id_b = str(obj_b.unique_identifier)
+
+        payload = locate.LocateRequestPayload()
+        e._logger.reset_mock()
+        response_payload = e._process_locate(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Locate"
+        )
+
+        self.assertEqual(
+           len(response_payload.unique_identifiers),
+           2
+        )
+        self.assertEqual(
+            id_a,
+            response_payload.unique_identifiers[0].value
+        )
+        self.assertEqual(
+            id_b,
+            response_payload.unique_identifiers[1].value
+        )
```

Comment:
These last two asserts should be changed to `assertIn` checks. There's no guarantee that SQLAlchemy will return the objects in the order they were added.